src/mbgl/map/map.cpp

```diff
@@ -727,44 +625,30 @@ void Map::updateAnnotation(AnnotationID id, const Annotation& annotation) {
 
 void Map::removeAnnotation(AnnotationID annotation) {
     impl->annotationManager.removeAnnotation(annotation);
-    impl->onUpdate(Update::AnnotationStyle | Update::AnnotationData);
+    impl->onUpdate(Update::AnnotationData);
 }
 
 #pragma mark - Feature query api
 
 std::vector<Feature> Map::queryRenderedFeatures(const ScreenCoordinate& point, const RenderedQueryOptions& options) {
```

Comment:
@jfirebaugh I moved the feature querying to the renderer and used it directly on platforms where we are going to keep synchronous rendering and through the renderer frontend implementation for the others.

Still need to move the `map/query.hpp` header to `render/`